core/3d/CCAnimation3D.cpp

```diff
@@ -113,8 +114,11 @@ bool Animation3D::init(const Animation3DData& data)
 
         if (iter.second.empty())
             continue;
-        std::vector<float> keys;
-        std::vector<float> values;
+
+        keys.clear();
+        values.clear();
+        keys.reserve(iter.second.size());
+        values.reserve(iter.second.size() * 3);
```

Comment:
This change make code less readable. 
What do you think about following change
```
        std::vector<float> keys;
        std::vector<float> values;
        keys.reserve(iter.second.size());
        values.reserve(iter.second.size() * 3);
```
If this performance improvement probably ~2x faster then reserve+emplace_back will be something like
```
std::vector<float> keys(iter.second.size());
std::transform(iter.second.begin(), iter.second.end(), keys.begin(), [](const auto& keyIter){return keyIter._time;});
```

Reply:
> This change make code less readable. What do you think about following change
> 
> ```
>         std::vector<float> keys;
>         std::vector<float> values;
>         keys.reserve(iter.second.size());
>         values.reserve(iter.second.size() * 3);
> ```
> 
> If this performance improvement probably ~2x faster then reserve+emplace_back will be something like
> 
> ```
> std::vector<float> keys(iter.second.size());
> std::transform(iter.second.begin(), iter.second.end(), keys.begin(), [](const auto& keyIter){return keyIter._time;});
> ```

And keys, values still could be reuse for 3 for-loops to avoid GC alloc